app/controllers/course_summaries_controller.rb

```diff
@@ -15,4 +15,116 @@ def populate
   def get_marking_scheme_details
     redirect_to url_for(controller: 'marking_schemes', action: 'populate')
   end
+
+  def download_csv_grades_report
+    csv_string = CSV.generate do |csv|
+      # Populates the entire CSV with headers, students and their marks
+      populate_students_and_marks(csv)
+    end
+    course_name = "#{COURSE_NAME}"
+    course_name_underscore = course_name.squish.downcase.tr(" ", "_")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.